types/parse/index.d.ts

```diff
@@ -29,7 +29,14 @@
 /// <reference path="node.d.ts" />
 /// <reference path="react-native.d.ts" />
 
-import { EventEmitter } from 'events';
+import { EventEmitter } from "events";
+
+/**
+ * Definitions for save and set forbid undefined values while keeping Intellisense working
+ * See https://github.com/DefinitelyTyped/DefinitelyTyped/blob/3bdadbf9583c2335197c7e999b9a30880e055f62/types/react/index.d.ts#L482
+ */
+type SafeAttributes<T, K extends keyof T> = Pick<Required<T>, K> | Required<T>;
```

Comment:
@rdhelms Sorry I wasn't clearer in my explanation then, this is also the approach I took, with a link to the React types

Reply:
Except we need to use `Required` in case some attributes are "optional"

Reply:
Do we really want to start preventing setting undefined in all cases though? That's not what the current behavior is, and it's not what the original intent was, and you mentioned that it may not be the job of the types package to enforce that.

Reply:
If you're on board with allowing `undefined` for cases where the attributes are specifically typed as either `Partial` or properties with `undefined`, I would vote to remove these helper types and just do `Pick<T, K> | T` at the `save` and `set` definitions.

Reply:
Agree, sounds like a good compromise. I pinged you here because I felt bad I didn't properly explain that the `Pick<T, K> | T` trick was already part of the PR, that's what I meant when I said "keep IntelliSense working"